locator/types.hh

```diff
@@ -31,6 +31,7 @@ struct endpoint_dc_rack {
     bool operator==(const endpoint_dc_rack&) const = default;
 };
 
-using dc_rack_fn = seastar::noncopyable_function<std::optional<endpoint_dc_rack>(inet_address)>;
+template <typename NodeId>
```

Comment:
I'd add a concept defining the types, but since it goes away in the end, no need to bother.